pkgs/build-support/rust/carnix.nix

```diff
@@ -461,690 +456,1053 @@ rec {
     libName = "build";
     inherit dependencies buildDependencies features;
   };
-  aho_corasick_0_6_3 = f: aho_corasick_0_6_3_ rec {
-    dependencies = [ (memchr_1_0_2 f) ];
-  };
-  aho_corasick_0_6_3_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-        memchr_1_0_2.default.from_aho_corasick_0_6_3__default_ = true;
-      }))
-    [ memchr_1_0_2_features ];
-  ansi_term_0_10_2 = f: ansi_term_0_10_2_ rec {};
-  ansi_term_0_10_2_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-      }))
-    [ ];
-  atty_0_2_3 = f: atty_0_2_3_ rec {
-    dependencies = (if kernel == "redox" then [ (termion_1_5_1 f) ] else [])
-      ++ (if (kernel == "linux" || kernel == "darwin") then [ (libc_0_2_33 f) ] else [])
-      ++ (if kernel == "windows" then [ (kernel32_sys_0_2_2 f) (winapi_0_2_8 f) ] else []);
-  };
-  atty_0_2_3_features = features:
-    lib.lists.foldl' (features: f: f features)
-      (lib.attrsets.recursiveUpdate features (rec {
-        kernel32_sys_0_2_2.default.from_atty_0_2_3__default_ = true;
-        libc_0_2_33.default.from_atty_0_2_3__default_ = false;
-        termion_1_5_1.default.from_atty_0_2_3__default_ = true;
-        winapi_0_2_8.default.from_atty_0_2_3__default_ = true;
-      }))
-    [ termion_1_5_1_features libc_0_2_33_features kernel32_sys_0_2_2_features winapi_0_2_8_features ];
-  backtrace_0_3_4 = f: backtrace_0_3_4_ rec {
-    dependencies = [ (cfg_if_0_1_2 f) (rustc_demangle_0_1_5 f) ]
+  aho_corasick_0_6_3 = { features?(aho_corasick_0_6_3_features {}) }: aho_corasick_0_6_3_ {
+    dependencies = [ (memchr_1_0_2 { features = features; }) ];
+  };
+  aho_corasick_0_6_3_features = f:
+    builtins.deepSeq f (lib.lists.foldl' (features: fun: fun features)
+      (lib.attrsets.recursiveUpdate f (rec {
+        aho_corasick_0_6_3.default = (f.aho_corasick_0_6_3.default or true);
+        memchr_1_0_2.default = true;
+      })
+    )
+    [ memchr_1_0_2_features ]);
+  ansi_term_0_10_2 = { features?(ansi_term_0_10_2_features {}) }: ansi_term_0_10_2_ {};
+  ansi_term_0_10_2_features = f:
+    builtins.deepSeq f (lib.lists.foldl' (features: fun: fun features)
+      (lib.attrsets.recursiveUpdate f (rec {
```

Comment:
Can this whole thing become a new function instead of pasting this for every crate?

```
builtins.deepSeq f (lib.lists.foldl' (features: fun: fun features) (lib.attrsets.recursiveUpdate f)
```

Reply:
Good idea. I'll do it now.